### ISSUE TITLE
`type-annotation-spacing`: options to treat newline separately or ignore certain type

### ISSUE THREAD

**Issue by n0099** [eslint-stylistic/eslint-stylistic#249]
### Clear and concise description of the problem

Example from https://github.com/typescript-eslint/typescript-eslint/issues/5305
- with the default option `{}`:
```ts
const propertyWithLongNameWhichExceedMaxLen
  : string = 'test';
  ^
Unexpected space before the ':'  @stylistic/type-annotation-spacing
```
```ts
const propertyWithLongNameWhichExceedMaxLen
: string = 'test';
^
Unexpected space before the ':'  @stylistic/type-annotation-spacing
```
```ts
const propertyWithLongNameWhichExceedMaxLen:
string = 'test';
// pass
```
- with option `{ overrides: { returnType: { before: true, after: true } } }`:
```ts
const propertyWithLongNameWhichExceedMaxLen
  : string = 'test';
//pass
```
```ts
const propertyWithLongNameWhichExceedMaxLen
: string = 'test';
//pass
```
```ts
const propertyWithLongNameWhichExceedMaxLen:
                                           ^
Expected a space before the ':'.(@stylistic/type-annotation-spacing)
string = 'test';
```
```ts
const propertyWithLongNameWhichExceedMaxLen: string = 'test';
                                           ^
Expected a space before the ':'.(@stylistic/type-annotation-spacing)
```
```ts
const propertyWithLongNameWhichExceedMaxLen : string = 'test';
// pass
```
- with option `{ overrides: { returnType: { before: true, after: false } } }`:
```ts
const propertyWithLongNameWhichExceedMaxLen
  : string = 'test';
  ^
Unexpected space after the ':'  @stylistic/type-annotation-spacing
```
```ts
const propertyWithLongNameWhichExceedMaxLen
: string = 'test';
^
Unexpected space after the ':'  @stylistic/type-annotation-spacing
```
```ts
const propertyWithLongNameWhichExceedMaxLen:
                                           ^
Unexpected space after the ':'  @stylistic/type-annotation-spacing
string = 'test';
```
```ts
const propertyWithLongNameWhichExceedMaxLen: string = 'test';
                                           ^
Unexpected space after the ':'  @stylistic/type-annotation-spacing
```
```ts
const propertyWithLongNameWhichExceedMaxLen : string = 'test';
                                            ^
Unexpected space after the ':'  @stylistic/type-annotation-spacing
```
```ts
const propertyWithLongNameWhichExceedMaxLen :string = 'test';
// pass
```

### Suggested solution

Add options to treat newline separately with space by each override.

### Alternative

Or allow complete ignoring certain type like `returnType` in this example.

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [ ] If this feature request is accepted, I am willing to submit a PR to fix this issue
